block-storage/src/main/scala/coop/rchain/blockstorage/FileLMDBIndexBlockStore.scala

```diff
@@ -23,17 +23,28 @@ import org.lmdbjava.DbiFlags.MDB_CREATE
 import org.lmdbjava._
 
 import scala.collection.JavaConverters._
-import scala.ref.WeakReference
 import scala.util.control.NonFatal
 import scala.util.matching.Regex
 
 class FileLMDBIndexBlockStore[F[_]: Monad: Sync: Log] private (
     lock: Semaphore[F],
     env: Env[ByteBuffer],
     index: Dbi[ByteBuffer],
+    storagePath: Path,
+    checkpointsDir: Path,
     blockMessageRandomAccessFileRef: Ref[F, RandomAccessFile],
-    checkpointsRef: Ref[F, List[Checkpoint]]
+    checkpointsRef: Ref[F, Map[Int, Checkpoint]],
```

Comment:
It looks to me that all those `Ref`s are used within `lock.withPermit` blocks. Since this prevents races, it would make more sense to put all of them inside a single `MonadState` object